pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -432,49 +432,104 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
         nonReentrant
         returns (uint256 issuedShares)
     {
+        if (_bufferAssets[wrappedToken] != address(0)) {
+            revert BufferAlreadyInitialized(wrappedToken);
+        }
+
         address underlyingToken = wrappedToken.asset();
 
         if (underlyingToken == address(0)) {
             // Should never happen, but a malicious wrapper could return the zero address and cause the buffer
             // initialization code to run more than once.
-            revert InvalidUnderlyingTokenAsset();
+            revert InvalidUnderlyingToken();
         }
 
+        // Register asset of wrapper, so it cannot change.
+        _bufferAssets[wrappedToken] = underlyingToken;
+
+        // Take debt for initialization assets.
+        _takeDebt(IERC20(underlyingToken), amountUnderlyingRaw);
+        _takeDebt(wrappedToken, amountWrappedRaw);
+
+        // Update buffer balances.
+        bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
+        bufferBalances = PackedTokenBalance.toPackedBalance(
+            bufferBalances.getBalanceRaw() + amountUnderlyingRaw,
+            bufferBalances.getBalanceDerived() + amountWrappedRaw
+        );
+
+        _bufferTokenBalances[wrappedToken] = bufferBalances;
+
         // Amount of shares to issue is the total underlying token that the user is depositing.
         issuedShares = wrappedToken.convertToAssets(amountWrappedRaw) + amountUnderlyingRaw;
 
-        if (_bufferAssets[wrappedToken] == address(0)) {
-            // Buffer is not initialized yet, so we initialize it.
+        _ensureMinimumTotalSupply(issuedShares);
```

Comment:
The flow here is a bit funky

```solidity
_ensureMinimumTotalSupply(issuedShares); // checks issuedShares > minimum

issuedShares -= _MINIMUM_TOTAL_SUPPLY; // subtracts minimum from issued shares
_mintBufferShares(wrappedToken, sharesOwner, issuedShares); // internally calls _ensureMinimumTotalSupply with the new smaller amount
```

We're calling `_ensureMinimumTotalSupply` twice with different numbers
- When calling `_mintBufferShares`, assuming the current supply is greater than the minimum the `newTotalSupply` can never be less than the minimum.
- The only place we need to check for the minimum is on init, no?
- If the argument is an additional security layer, why are you not checking for minimum on remove. This is arguably the more important place for this extra check.